client/state/sharing/keyring/actions.js

```diff
@@ -48,3 +49,30 @@ export const deleteKeyringConnection = ( connection ) => ( dispatch ) => dispatc
 	type: KEYRING_CONNECTION_DELETE,
 	connection,
 } );
+
+/**
+ * Triggers a network request to delete a Keyring connection from the server-side.
+ *
+ * @param  {Object} connection         Connection to be deleted.
+ * @param  {Number} connection.ID      ID of the connection to be deleted.
+ * @param  {String} connection.label   Name of the service that was connected.
+ * @return {Function}                  Action thunk
+ */
+export function deleteStoredKeyringConnection( connection ) {
+	return ( dispatch ) =>
+		wpcom.undocumented().deletekeyringConnection( connection.ID )
+			.then( () => deleteKeyringConnection( connection )( dispatch ) )
+			.catch( ( error ) => {
+				if ( error && 404 === error.statusCode ) {
+					// If the connection cannot be found, we infer that it must have been deleted since the original
+					// connections were retrieved, so pass along the cached connection.
+					deleteKeyringConnection( connection )( dispatch );
```

Comment:
Aside: Why is `deleteKeyringConnection` a thunk? It's not doing anything asynchronous. Makes this line more complex than it should be.